base/sparse/higherorderfns.jl

```diff
@@ -871,11 +874,11 @@ promote_containertype(::Type{AbstractSparseArray}, ::Type{Tuple}) = Array
 
 # broadcast[!] entry points for combinations of sparse arrays and other (scalar) types
 @inline function broadcast_c{N}(f, ::Type{AbstractSparseArray}, mixedargs::Vararg{Any,N})
-    parevalf, passedargstup = capturescalars(f, mixedargs)
+    parevalf, passedargstup = capturescalars(f, map(_sparsifystructured, mixedargs))
```

Comment:
I have added a `capturedscalars` method that should take care of this for now until we find a better alternative (I have thought of having a method that iterates over columns for `AbstractArray`s in general and doing something like `_broadcast!` with `_broadcast_getindex!` so we don't need this mechanism, but that would have to wait)